### ISSUE TITLE
Blazor: NavigationManager RegisterLocationChangingHandler and LocationChanged never are invoked

### ISSUE THREAD

**Issue by Cephei** [dotnet/aspnetcore#44365]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I am attempting a proof-of-concept using:
- dotnet 7 RC1 
- `Microsoft.AspNetCore.Components.CustomElements`
- Blazor Server
- React

It appears that the `NavigationManager` is unable to listen to the location change events that occur from React. See the screenshot below; the `OnLocationChanging(LocationChangingContext context)` method is never invoked. I've tried everything I can think of, but I'm a complete novice to Blazor. Perhaps I overlooked something.

<img width="745" alt="Screen Shot 2022-10-04 at 8 27 58 PM" src="https://user-images.githubusercontent.com/1054639/193897048-7c777548-8d5e-4fd0-86e9-f56878d18188.png">

### Expected Behavior

NavigationManager should be listening to client-side browser events.

### Steps To Reproduce

In my `index.html` I have include the following:
```js
<script src="%REACT_APP_EID_BASE_URL%/_content/Microsoft.AspNetCore.Components.CustomElements/BlazorCustomElements.js"></script>
<script src="%REACT_APP_EID_BASE_URL%/_framework/blazor.server.js" autostart="false"></script>
<script>
      Blazor.start({
        configureSignalR: function (builder) {
          builder.withUrl("%REACT_APP_EID_BASE_URL%/_blazor");
        }
      });
</script>
```

### .NET Version

7.0.100-rc.1.22431.12

### Anything else?

I've also attempted this with the `LocationChanged` event, but to the same result.
In React, all I'm doing is using the `react-router-dom` + `useNavigate` hook. The application itself navigates just fine. I just can't listen to the location changes in the blazor component.

I should also note, that `CustomElements` seems to be working otherwise. The razor component `OnInitializedAsync` is invoked just fine. This could be really remarkable tech in dotnet 7.

**Comment by mkArtakMSFT** [comment]
@MackinnonBuck can you please look into this? Thanks!

**Comment by MackinnonBuck** [comment]
Unfortunately, there's probably not much we can do to make this scenario work. Both `react-router-dom` and Blazor's JavaScript-side navigation management are fairly intrusive mechanisms that assume almost complete control over the site's navigation, so they're bound to be incompatible by nature.

**Comment by javiercn** [comment]
The only way I can think this can be done is by hooking into the react router to detect when it is handling changes and to propagate those to Blazor but I do not believe we expose a direct way to achieve it.

**Comment by Cephei** [comment]
I appreciate the response guys. I came to the same conclusion after digging further into React's support for webcomponents and custom elements. I started down a path of doing something like:
```
useEffect(() => {
    // update Blazor component with new location
  }, [location])
```

But to be honest this became a bit too complex for the time I have allocated to prototype. I'm sure it could probably be done, but I think it may also be overkill than just writing a native react & angular component for what I'm intending to accomplish.

Thanks for taking a look at it 🍻
